http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/RuntimeV2ServiceInterp.scala

```diff
@@ -312,12 +314,31 @@ class RuntimeV2ServiceInterp[F[_]: Parallel](config: RuntimeServiceConfig,
           )
       }
 
-      // We only update IP to properly after the VM is created in WSM. Hence, if IP is not defined, we don't need to pass the VM resourceId
-      // to back leo, where it'll trigger a WSM call to delete the VM
-      wsmVMResourceSamId = runtime.hostIp.map(_ => WsmControlledResourceId(UUID.fromString(runtime.internalId)))
-      // If there's no controlled resource record, that means we created the DB record in front leo but failed somewhere in back leo (possibly in polling WSM)
-      // This is non-fatal, as we still want to allow users to clean up the db record if they have permission.
-      // We must check if they have permission other ways if we did not get an ID back from WSM though
+      // get wsm api
+      wsmAzureResourceApi = wsmClientProvider.getControlledAzureResourceApi(userInfo.accessToken.token)
+      wsmResourceId = WsmControlledResourceId(UUID.fromString(runtime.internalId))
+
+      // if the vm is found in WSM and has a deletable state,
+      // then the resourceId is passed to back leo to make the delete call to WSM
+      // (state can be BROKEN, CREATING, DELETING, READY, UPDATING or NULL)
+      deletableStatus = List("BROKEN", "READY")
+
+      wsmVMResourceSamId = Try(wsmAzureResourceApi.getAzureVm(workspaceId.value, wsmResourceId.value)) match {
```

Comment:
Friendly Scala tip.. the blocking WSM call should probably be "lifted" into the F type and chained in the for-comp since it's a side effect. Also the `log` statements below are also `F`s but not chained to anything, so they probably won't execute as is. One way to write this could be:
```
      attempt <- F.delay(wsmAzureResourceApi.getAzureVm(workspaceId.value, wsmResourceId.value)).attempt
      wsmVMResourceSamId <- attempt match {
        case Right(result) =>
          val vmState = result.getMetadata.getState.getValue
          val res = if (deletableStatus.contains(vmState)) Some(wsmResourceId) else None
          log
            .info(
              s"Runtime ${runtimeName.asString} with resourceId ${wsmResourceId.value} has a state of $vmState in WSM"
            )
            .as(res)
        case Left(e) =>
          log
            .info(
              s"No wsm record found for runtime ${runtimeName.asString} No-op for wsmDao.deleteVm, ${e.getMessage}"
            )
            .as(None)
      }
```

Reply:
ohh thank you! Yeah I realized that those logs weren't executing, that makes sense